### PR TITLE
Rename SHA3 opcode to KECCAK256

### PR DESCRIPTION
**Pull request by Demuirgos**
Fixes Closes Resolves #

_Please choose one of the keywords above to refer to the issue this PR solves followed by the issue number (e.g. Fixes #000). If no issue number, remove the line. Also, remove everything marked optional that is not applicable. Remove this note after reading._

## Changes

- Align with other clients with regards to opcode 0x20 naming : SHA3 -> KECCAK256

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [x] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [ ] Yes
- [x] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

**Review by smartprogrammer93 — APPROVED** [review]
As discussed this reflects the execution spec and how other clients name the opcode

File: src/Nethermind/Nethermind.Evm.Test/Sha3Tests.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public void Spin_sha3()
                 .Op(Instruction.JUMPDEST)
                 .PushData(32)
                 .PushData(0)
-                .Op(Instruction.SHA3)
+                .Op(Instruction.KECCAK256)
```
Comment (rubo) [inline src/Nethermind/Nethermind.Evm.Test/Sha3Tests.cs:36]: Rename `Sha3Tests` to `Keccak256Tests`

**Review by rubo — CHANGES_REQUESTED** [review]

**Review by benaadams — APPROVED** [review]

**Comment by LukaszRozmej** [comment]
Not sure but the reason for the naming might have been compatibility with `Parity`/`OpenEthereum`. maybe this will break something in traces from `trace` module?

**Comment by rubo** [comment]
> maybe this will break something in traces from `trace` module?

@LukaszRozmej If any old-style tracing compatibility is broken, then it should be addressed in that particular tracer itself. This should've been renamed long ago, the choice of `SHA3` was just historical and not for compatibility reasons. This already has already been discussed a year ago.

**Comment by rubo** [comment]
@Demuirgos `Update_known_chain_sizes` test fails for Goerli. Not sure what's wrong there.

**Review by MarekM25 — APPROVED** [review]

**Review by rubo — APPROVED** [review]

**Pull request merged**